src/main/java/org/ow2/proactive/addons/ldap_query/LDAPClient.java

```diff
@@ -0,0 +1,232 @@
+            throw new IllegalArgumentException("The missed argument for LDAPClient, variable name: " + variableName);
+        }
+        String varValue = getAsString(taskVariablesMap, variableName);
+        try {
+            this.getClass().getDeclaredField(variableName).set(this, varValue);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("The variable name is wrong:" + variableName, e);
+        }
+    }
+
+    private String getAsString(Map<String, Serializable> map, String argFrom) {
+        return (String) map.get(argFrom);
+    }
+
+    private String[] splitAttributes(String attrList) {
+        String[] splittedAttr = new String[0];
+        if (attrList != null && !attrList.trim().isEmpty()) {
+            splittedAttr = attrList.split(REGEX_LIST_SEPARATOR);
+
+        }
+        return splittedAttr;
+    }
+
+    public String searchQueryLDAP() {
+        NamingEnumeration results = null;
+        ObjectMapper mapper = new ObjectMapper();
+        Response response;
+        boolean allAttrs = false;
+        String resultOutput = new String();
+        List attributesList = new LinkedList();
+        HashSet<String> attrReturn = new HashSet();
+
+        String[] attributesToReturn = splitAttributes(ldapSelectedAttributes);
+
+        if (attributesToReturn.length == 0) {
+            allAttrs = true;
+        } else {
+            attrReturn = new HashSet<>(Arrays.asList(attributesToReturn));
+        }
+
+        try {
+            ldapConnection = LDAPConnectionUtility.connect(ldapUrl, ldapUsername, ldapPassword);
+            SearchControls controls = new SearchControls();
+            controls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+            results = ldapConnection.search(ldapSearchBase, ldapSearchFilter, controls);
+
+            //iterate throw all attributes in the result of search query
+            while (results.hasMore()) {
+                SearchResult searchResult = (SearchResult) results.next();
+                Attributes attributes = searchResult.getAttributes();
+                Map<String, String> attrMap = new HashMap();
+                for (NamingEnumeration ae = attributes.getAll(); ae.hasMore();) {
+                    Attribute attr = (Attribute) ae.next();
+                    String attrId = attr.getID();
+                    if ((!allAttrs && attrReturn.contains(attrId)) || allAttrs) {
+                        attrMap.put(attrId, attr.get().toString());
+                    }
+                }
+                if (!attrMap.isEmpty()) {
+                    attributesList.add(attrMap);
+                }
+            }
+            response = new LDAPResponse("Ok", attributesList);
+        } catch (Exception e) {
+            response = new ErrorResponse("Error", e.toString());
+        } finally {
+            if (results != null) {
+                try {
+                    results.close();
+                } catch (Exception e) {
```

Comment:
Please do something with the exception. As least, log or print to standard err if the purpose is to be used from a ProActive Task.